data_steward/retraction/create_deid_map.py

```diff
@@ -0,0 +1,172 @@
+    for d in deid_datasets:
+        release = d.split('_')[0]
+        prefix = release[1:]
+        combined = prefix + '_combined'
+
+        if combined in all_datasets:
+            new_row = pd.Series({
+                'deid_dataset': d,
+                'combined_dataset': combined
+            })
+            deid_and_combined_datasets_df = deid_and_combined_datasets_df.append(
+                new_row, ignore_index=True)
+
+    return deid_and_combined_datasets_df
+
+
+def check_if_deid_map_exists(project_id, dataset):
+    """
+    Checks if _deid_map table exists in the given dataset.
+    :param project_id: bq name of project_id
+    :param dataset: bq name of dataset
+    :return: returns True, False or 'rename required' which had 'deid_map' table instead of '_deid_map'
+    """
+    table_info_df = utils.bq.get_table_info_for_dataset(project_id, dataset)
+    column_list = table_info_df['table_name'].tolist()
+
+    if '_deid_map' in column_list:
+        return consts.SKIP
+    elif 'deid_map' in column_list:
+        return consts.RENAME
+    else:
+        return consts.CREATE
+
+
+def rename_deid_map_table_query(project_id, dataset):
+    return consts.RENAME_DEID_MAP_TABLE_QUERY.format(project=project_id,
+                                                     dataset=dataset)
+
+
+def create_deid_map_table_query(project, dataset):
+    return consts.CREATE_DEID_MAP_TABLE_QUERY.format(project=project,
+                                                     dataset=dataset)
+
+
+def create_deid_map_table_queries(project):
+    """
+    Creates a query list to run to create or rename _deid_map tables in each combined dataset that has a deid dataset
+    :param project: bq name of project_id
+    :return: list of queries to run
+    """
+    deid_and_combined_df = get_combined_datasets_for_deid_map(project)
+    combined_datasets_list = deid_and_combined_df['combined_dataset'].to_list()
+    # remove duplicate combined datasets in list
+    combined_datasets_list = list(set(combined_datasets_list))
+    queries = []
+    for dataset in combined_datasets_list:
+        check = check_if_deid_map_exists(project, dataset)
+        if check == 'skip':
+            continue
+        if check == 'rename':
+            queries.append(rename_deid_map_table_query(project, dataset))
+        if check == 'create':
+            queries.append(create_deid_map_table_query(project, dataset))
+    return queries
+
+
+def run_queries(queries):
+    """
+    Function that will run the queires to create '_deid_map' tables
+
+    :param queries: list of queries
+    """
+    query_job_ids = []
+    for query_dict in queries:
+        logging.info('Creating or renaming _deid_map table with query: %s' %
+                     (query_dict['query']))
+        job_results = bq_utils.query(q=query_dict['query'], batch=True)
+        rows_affected = job_results['numDmlAffectedRows']
+        logging.info('%s rows written to _deid_map table' % rows_affected)
```

Comment:
I'm not sure if `numDmlAffectedRows` is returned when a `CREATE OR REPLACE TABLE` is used, let me know if you've tested it

Reply:
you're correct, this is removed.